### PR TITLE
[SPARK-44494][INFRA] Use `minikube` v1.30.1 for `k8s-integration-tests`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr change to use `minikube` v1.30.1 for `k8s-integration-tests` on GitHub Action, this is a temporary solution.
This PR also leaves a TODO:

- SPARK-44495: Resume to use the latest minikube for `k8s-integration-tests` on GitHub Action

### Why are the changes needed?
Restore `k8s-integration-tests` GA testing

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- `k8s-integration-tests` test pass on GitHub Action

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -1009,7 +1009,8 @@ jobs:
       - name: start minikube
         run: |
           # See more in "Installation" https://minikube.sigs.k8s.io/docs/start/
-          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+          # curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+          curl -LO https://storage.googleapis.com/minikube/releases/v1.30.1/minikube-linux-amd64
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:1013]: The latest version is v1.31.0 and the k8s-it failed last 2 days

**Comment by LuciferYang** [comment]
@Yikun @HyukjinKwon I haven't found the root cause of the failure yet. This pr is trying to restore GA using last successful minikube version
first. WDYT?

**Comment by HyukjinKwon** [comment]
Yeah, if it fixes the build it's good

**Comment by LuciferYang** [comment]
image

seems ok, let me pr description and leave a TODO

**Review by panbingkun — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for recovering CI, @LuciferYang .

**Comment by dongjoon-hyun** [comment]
Merged to master/3.5. If we need this to branch-3.4 and older, please feel free to backport it, @LuciferYang .

**Comment by LuciferYang** [comment]
@dongjoon-hyun OK ~

**Pull request closed without merging**